Add debugging info in pv/pvc protection e2e tests

oomichi opened the pull request:
**What this PR does / why we need it**:

When facing an issue which is due to lack of PV/PVC Protection
finalizer on the e2e tests, the error message is just like
```
  Expected
      <bool>: false
  to be true
```
Actually we cannot understand what happened during the tests.
This adds more debugging info on the tests for easy debugging.

This is for debugging the issue https://github.com/kubernetes/kubernetes/issues/67519

**Release note**: NONE

oomichi commented:
/release-note-none

neolit123 commented:
/sig storage

oomichi commented:
/assign @gnufied

gnufied commented on test/e2e/storage/pvc_protection.go:
```diff
@@ -63,7 +65,7 @@ var _ = utils.SIGDescribe("PVC Protection", func() {
 		pvc, err = client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(pvc.Name, metav1.GetOptions{})
 		Expect(err).NotTo(HaveOccurred(), "While getting PVC status")
-		Expect(slice.ContainsString(pvc.ObjectMeta.Finalizers, volumeutil.PVCProtectionFinalizer, nil)).To(BeTrue())
+		Expect(slice.ContainsString(pvc.ObjectMeta.Finalizers, volumeutil.PVCProtectionFinalizer, nil)).To(BeTrue(), fmt.Sprintf("PVC Protection finalizer(%v) is not set in %v", volumeutil.PVCProtectionFinalizer, pvc.ObjectMeta.Finalizers))
```
IIRC there is no need to use `fmt.Sprintf` here. `Expect` already supports a variadic/printf style of formatting messages.

oomichi replied:
Thanks for your advice, I see.
I will try it on next update :-)

Commit:
Add debugging info in pv/pvc protection e2e tests

When facing an issue which is due to lack of PV/PVC Protection
finalizer on the e2e tests, the error message is just like

  Expected
      <bool>: false
  to be true

Actually we cannot understand what happened during the tests.
This adds more debugging info on the tests.

oomichi commented:
/test pull-kubernetes-e2e-gce-100-performance

oomichi commented:
@gnufied Hi, thanks for your review. Is current one good for you?

shu-mutou commented:
/lgtm

oomichi commented:
For getting approval

/assign @spiffxp

spiffxp commented:
/lgtm
/approve

oomichi commented:
/test pull-kubernetes-e2e-kops-aws

oomichi commented:
/test pull-kubernetes-integration
/test pull-kubernetes-e2e-gce-100-performance
